components/form/HookOption.vue

```diff
@@ -0,0 +1,271 @@
+            :label="t('workload.container.lifecycleHook.exec.command.label')"
+            :placeholder="t('workload.container.lifecycleHook.exec.command.placeholder')"
+          />
+        </div>
+      </div>
+    </template>
+
+    <template v-if="selectHook === 'httpGet'">
+      <template>
+        <h4>{{ t('workload.container.lifecycleHook.httpGet.title') }}</h4>
+        <slot name="httpGetOption">
+          <div class="var-row">
+            <div v-for="option in httpGetOptions" :key="option.vKey">
+              <LabeledInput
+                v-if="option.value === 'port'"
+                v-model.number="lifecycle.httpGet[option.value]"
+                type="number"
+                :label="option.label"
+                :placeholder="option.placeholder"
+                :mode="mode"
+              />
+              <LabeledInput
+                v-else
+                v-model="lifecycle.httpGet[option.value]"
+                :label="option.label"
+                :placeholder="option.placeholder"
+                :mode="mode"
+              />
+            </div>
+          </div>
+        </slot>
+      </template>
+
+      <template>
+        <h4>{{ t('workload.container.lifecycleHook.httpHeaders.title') }}</h4>
+        <template>
+          <div v-for="(header, index) in lifecycle.httpGet.httpHeaders" :key="header.vKey" class="var-row">
+            <LabeledInput
+              v-model="lifecycle.httpGet.httpHeaders[index].name"
+              :label="t('workload.container.lifecycleHook.httpHeaders.name.label')"
+              :placeholder="t('workload.container.lifecycleHook.httpHeaders.name.placeholder')"
+              class="single-value"
+              :mode="mode"
+            />
+            <LabeledInput
+              v-model="lifecycle.httpGet.httpHeaders[index].value"
+              :label="t('workload.container.lifecycleHook.httpHeaders.value.label')"
+              :placeholder="t('workload.container.lifecycleHook.httpHeaders.value.placeholder')"
+              class="single-value"
+              :mode="mode"
+            />
+            <button
+              v-if="!isView"
+              type="button"
+              class="btn role-link"
+              :disabled="mode==='view'"
+              @click.stop="removeHeader(index)"
+            >
+              <t k="generic.remove" />
+            </button>
+          </div>
+        </template>
+        <div>
+          <button
+            v-if="!isView"
+            type="button"
+            class="btn role-link mb-20"
+            :disabled="mode === 'view'"
+            @click.stop="addHeader"
+          >
+            Add Header
+          </button>
+        </div>
+      </template>
+    </template>
+
+    <template v-if="selectHook === 'tcpSocket'">
+      <h4>{{ t('workload.container.lifecycleHook.tcpSocket.title') }}</h4>
+      <template>
+        <div class="var-row">
+          <div v-for="option in tcpSocketOptions" :key="option.vKey">
```

Comment:
Functionality is good here but I'm not sure what we're gaining with the `tcpSocketOptions` array either since there's still two inputs in the template; why use `v-for` here instead of just adding `port` and `host` labeledinputs directly?

Reply:
Made tcpSocket a template with two input fields (`port`, `host`) and removed `tcpSocketOptions`.